### PR TITLE
Newstyle csproj & netstandard

### PR DESCRIPTION
**Pull request by omsmith**
* Update to newstyle csproj
* Update to 4.7.2 to deal with dependency hell
* Update to netstandard 2.0 since that's the end goal anyway, and the LMS is on 4.7, so 4.7.2 isn't a good transition step

# TODO
- [x] Test Locally
- [x] ~Wait for Roslyn / VS to support netstandard 2.0 analyzers~ Switch to netstandard 1.3
- [x] Re-test locally

### CODE REVIEW

**Comment by j3parker** [comment]
Cool! Not upgrading the integration tests yet because that project will need more stuff (maybe a glob-based item remove for the spec tests + a glob embedded resource item?)

@mthjones knows of an exe that has done some pretty smart things migrating projects

**Comment by omsmith** [comment]
Just haven't tried.

**Comment by mthjones** [comment]
This is awesome. :) I'm curious what the dependency hell issues were with targeting net47 though.

The exe @j3parker is mentioning is [this migrator tool](https://github.com/hvanbakel/CsprojToVs2017). It takes care of so much of the grunt work.

**Comment by omsmith** [comment]
`Microsoft.CodeAnalysis.Something` depends on `System.ValueTuple` as a nuget project. `System.ValueTuple` was added to the framework in .NET 4.7 and so you've got competing assemblies for it. .NET 4.7.1 fixed this, somehow, I think by knowing to ignore the nuget assembly.

There's probably other things I could've done, like adding a `PackageReference` to ignore valuetuple or the like, but... uhg.

**Comment by mthjones** [comment]
Ah yeah... perhaps targetting `netstandard1.3` like [Roslynator](https://github.com/JosefPihrt/Roslynator/blob/master/src/Analyzers/Analyzers.csproj) does would work better? FWIW, Roslynator works perfectly as a `PackageReference` in VS for me (the current project doesn't).

**Comment by omsmith** [comment]
Yeah, in order to move forward today, have to target netstandard1.3 (and thus fix some things in the analyzer that aren't in netstandard1.3). 2.0 should be supported "soon".

File: src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj [hunk 1]
```diff
@@ -1,239 +1,29 @@
...
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
-    <None Include="tools\uninstall.ps1">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
-  </ItemGroup>
-  <ItemGroup>
-    <Analyzer Include="..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll" />
-    <Analyzer Include="..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll" />
+    <None Update="tools\*.ps1" CopyToOutputDirectory="Always" Pack="true" PackagePath="tools" />
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj]: @mthjones

Comment (mthjones) [reply src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj]: Ahhh, my bad. Github thought this diff was too large to render, so Ctrl-F failed me. :(

File: src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj [hunk 2]
```diff
@@ -1,239 +1,29 @@
...
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <TargetFramework>netstandard1.3</TargetFramework>
     <LangVersion>7.3</LangVersion>
+
+    <Title>D2L.CodeStyle.Analyzers</Title>
+    <Product>D2L.CodeStyle</Product>
+    <Description>D2L.CodeStyle analyzers</Description>
+    <Version>0.92.0</Version>
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/D2L.CodeStyle.Analyzers.csproj]: Nice

**Pull request merged**